pkg/karmadactl/join/join.go

```diff
@@ -150,6 +159,12 @@ func (j *CommandJoinOption) AddFlags(flags *pflag.FlagSet) {
 	flags.StringVar(&j.ClusterProvider, "cluster-provider", "", "Provider of the joining cluster. The Karmada scheduler can use this information to spread workloads across providers for higher availability.")
 	flags.StringVar(&j.ClusterRegion, "cluster-region", "", "The region of the joining cluster. The Karmada scheduler can use this information to spread workloads across regions for higher availability.")
 	flags.StringSliceVar(&j.ClusterZones, "cluster-zones", nil, "The zones of the joining cluster. The Karmada scheduler can use this information to spread workloads across zones for higher availability.")
+	flags.StringVar(&j.KarmadaAs, "karmada-as", "",
+		"Host cluster username to impersonate for the operation. User could be a regular user or a service account in a namespace.")
+	flags.StringArrayVar(&j.KarmadaAsGroups, "karmada-as-group", []string{},
+		"Host cluster group to impersonate for the operation, this flag can be repeated to specify multiple groups.")
+	flags.StringVar(&j.KarmadaAsUID, "karmada-as-uid", "",
+		"Host cluster UID to impersonate for the operation.")
```

Comment:
The karmada naming seems a bit strange. Is it possible to reconsider the following three naming methods?
```
as-username-on-host
as-group-on-host
as-uid-on-host

host-as
host-as-group
host-as-uid

as
as-group
as-uid
```
What would be a better name for it?

Reply:
I would prefer to the first group as it more clear and straightforward.
@grosser what do you think?
Sorry for giving the wrong directions at https://github.com/karmada-io/karmada/issues/4695#issuecomment-1989757658.

Reply:
- 1 and 2 seem fine, 3 is confusing
- `as-username-on-host` don't like this since it breaks the symmetry to kubectl
- prefer 2 in general since it means all options for the host stay neatly grouped together

thoughts ?

Reply:
I prefer 1, and also OK with 2.

> as-username-on-host don't like this since it breaks the symmetry to kubectl

Yes, we usually follow the symmetry of Kubernetes, which would make it easier for users to understand.
We all agree that 3 is confusing, so we can't 100% follow kubectl, so I tend to the one more straightforward.

Group 1 is more clear than Group 2. But no strong opinion.  You can make the call.

> prefer 2 in general since it means all options for the host stay neatly grouped together

+1

Reply:
I prefer 1, and also OK with 2.

> as-username-on-host don't like this since it breaks the symmetry to kubectl

Yes, we usually follow the symmetry of Kubernetes, which would make it easier for users to understand.
We all agree that 3 is confusing, so we can't 100% follow kubectl, so I tend to the one more straightforward.

Group 1 is more clear than Group 2. But no strong opinion.  You can make the call.

> prefer 2 in general since it means all options for the host stay neatly grouped together

+1